server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v71/MigrateWebhooksToWebhooksTable.java

```diff
@@ -0,0 +1,300 @@
+  }
+
+  private List<Long> insert(Context context, Webhook webhook) throws SQLException {
+
+    if (webhook.isValid()) {
+      context.prepareUpsert("insert into webhooks (uuid, name, url, organization_uuid, project_uuid, created_at, updated_at) values (?, ?, ?, ?, ?, ?, ?)")
+        .setString(1, uuidFactory.create())
+        .setString(2, webhook.name())
+        .setString(3, webhook.url())
+        .setString(4, webhook.organisationUuid())
+        .setString(5, webhook.projectUuid())
+        .setLong(6, webhook.createdAt())
+        .setLong(7, webhook.createdAt())
+        .execute()
+        .commit();
+    } else {
+      LOGGER.info("Unable to migrate inconsistent webhook (entry deleted from PROPERTIES) : " + webhook);
+    }
+
+    return webhook.rowIds();
+  }
+
+  private static String deleteRequest(List<Long> toBeDeleted) {
+    return new StringBuilder()
+      .append("DELETE from properties WHERE id in (")
+      .append(Joiner.on(',').join(toBeDeleted))
+      .append(")")
+      .toString();
+  }
+
+  private List<Webhook> extractGlobalWebhooksFrom(Context context, Map<String, PropertyRow> properties, String[] values) throws SQLException {
+    String defaultOrganizationUuid = defaultOrganizationUuidProvider.get(context);
+    return Arrays.stream(values)
+      .map(value -> new Webhook(
+        properties.get("sonar.webhooks.global." + value + ".name"),
+        properties.get("sonar.webhooks.global." + value + ".url"),
+        defaultOrganizationUuid, null))
+      .collect(toList());
+  }
+
+  private List<Webhook> extractProjectWebhooksFrom(Map<String, PropertyRow> properties, String[] values, Map<Long, String> projectsKeys) {
+    return Arrays.stream(values)
+      .map(value -> new Webhook(
+        properties.get("sonar.webhooks.project." + value + ".name"),
+        properties.get("sonar.webhooks.project." + value + ".url"),
+        null, projectsKeys))
+      .collect(toList());
+  }
+
+  private static class PropertyRow {
+
+    private final Long id;
+    private final String key;
+    private final Long resourceId;
+    private final String value;
+    private final Long createdAt;
+
+    public PropertyRow(long id, String key, Long resourceId, String value, Long createdAt) {
+      this.id = id;
+      this.key = key;
+      this.resourceId = resourceId;
+      this.value = value;
+      this.createdAt = createdAt;
+    }
+
+    public Long id() {
+      return id;
+    }
+
+    public String key() {
+      return key;
+    }
+
+    public Long resourceId() {
+      return resourceId;
+    }
+
+    public String value() {
+      return value;
+    }
+
+    public Long createdAt() {
+      return createdAt;
+    }
+
+    @Override
+    public String toString() {
+      return "{" +
+        "id=" + id +
+        ", key='" + key + '\'' +
+        ", resourceId=" + resourceId +
+        ", value='" + value + '\'' +
+        ", createdAt=" + createdAt +
+        '}';
+    }
+  }
+
+  private class Webhook {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make this a "static" inner class. ([squid:S2694](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2694))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2FMMF-647%2FSONAR-10345_manage_webhooks&issues=AWGP3mXmOl9lQhPgXgq7&open=AWGP3mXmOl9lQhPgXgq7)

[ISSUE_KEY]: AWGP3mXmOl9lQhPgXgq7